sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -420,6 +430,10 @@ impl InputSelection {
 
         self.validate_transitions()?;
 
+        // Only keep mana rewards for selected inputs
```

Comment:
But why do we have more to start with?

Reply:
We shouldn't, but the user could pass in whatever so I figured I'd check

Reply:
Wouldn't it be better to return an error in this case?

Reply:
I suppose

Reply:
done